rfd/0083-license-expiration-warnings.md

```diff
@@ -0,0 +1,153 @@
+TCTL - “tctl status”
+When the user uses the “tctl status” command the appropriate license warning will be displayed.
+After expiration all tsh and tctl commands will display the license expiry warning.
+
+### Example outputs
+
+The license warning portion of the outputs should be coloured red and orange respectively for the expired and unexpired warnings.
+
+```
+$ build/tsh status
+Profile URL:        https://localhost:9876
+Logged in as:       someUser
+Cluster:            someCluster
+Roles:              access, editor
+Logins:             someUser
+Valid until:        2022-08-12 21:57:19 +0100 IST [valid for 4h44m0s]
+Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty
+Your Teleport Enterprise Edition license will expire in 10 days. Please reach out to [licenses@goteleport.com](mailto:licenses@goteleport.com) to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+```
+$ build/tsh login --proxy=localhost:9876 --user someUser
+Profile URL:        https://localhost:9876
+Logged in as:       someUser
+Cluster:            localhost
+Roles:              access, editor
+Logins:             someUser
+Valid until:        2022-08-16 03:38:07 +0100 IST [valid for 12h0m0s]
+Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty
+
+Your Teleport Enterprise Edition license has expired. Please reach out to [licenses@goteleport.com](mailto:licenses@goteleport.com) to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+```
+$ build/tctl status -c /home/edward/teleport/teleport.yaml
+Cluster  localhost                                                               
+Version  10.1.2                                                                  
+host CA  never updated                                                           
+user CA  never updated                                                           
+db CA    never updated                                                           
+jwt CA   never updated                                                           
+CA pin   sha256:3d72102f020146d09ff400810f59f70b8163ebfe6ec1ecfb0b3b2a0c151592
+
+Your Teleport Enterprise Edition license has expired. Please reach out to [licenses@goteleport.com](mailto:licenses@goteleport.com) to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support.
+```
+
+
+CLI does not support snoozing the warnings as you can in the webUI. Which provides the functionality to disable the warnings temporarily.
+
+### Web UI UX
+
+Web UI will display license expiration banner on top of the page with the following rules:
+Show warning (yellow) to users N days prior to expiration, 1 day snooze
+Show error (red) to all users after expiration, no dismiss or snoozing available.
```

Comment:
Note for how the UI interprets severity integer values:
| Severity | Value | Color |
|---|---|---|
| low | 0 - 4 | blue |
| medium | 5 - 9 | yellow |
| high | 10 + | red |